gems/aws-sdk-core/lib/aws-sdk-core/log/param_filter.rb

```diff
@@ -6,40 +6,40 @@
-      # end
-
       def initialize(options = {})
-        @filters = Set.new(SENSITIVE + Array(options[:filter]))
+        @enabled = options[:filter_sensitive_params] != false
+        @additional_filters = options[:filter] || []
       end
 
-      def filter(value)
-        case value
-        when Struct, Hash then filter_hash(value)
-        when Array then filter_array(value)
-        else value
+      def filter(values, type = nil)
+        case values
+        when Struct, Hash then filter_hash(values, type)
+        when Array then filter_array(values, type)
+        else values
         end
       end
 
       private
 
-      def filter_hash(values)
+      def filter_hash(values, type)
+        filters = @additional_filters
+        if type.const_defined?('SENSITIVE')
```

Comment:
Check type nil since its the default on filter. eg: `if type && type.const_defined?`